addons/sale/models/sale_order_line.py

```diff
@@ -953,6 +957,12 @@ def _onchange_product_packaging_id(self):
                     },
                 }
 
+    @api.onchange('order_id')
+    def _onchange_order_id(self):
+        for line in self:
+            if line._origin.order_id:
+                line.product_id = False
+
```

Comment:
If was to avoid having a product at the same time as no order when creating a new SOL, because that leads to a traceback.
However, the sale team is against adding this onchange so it's not relevant anymore